src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -307,10 +307,15 @@ class PatternMatcher extends MiniPhaseTransform with DenotTransformer {thisTrans
       // TODO Deal with guards?
 
       def isSwitchableType(tpe: Type): Boolean = {
-        (tpe isRef defn.IntClass) ||
-        (tpe isRef defn.ByteClass) ||
-        (tpe isRef defn.ShortClass) ||
-        (tpe isRef defn.CharClass)
+        val actualTpe = tpe match {
+          case t @ AnnotatedType(inner, _) => inner
```

Comment:
What if the type has several annotations? Use `stripAnnots` instead (in fact, it might make sense to use `stripAnnots` in `isRef`, but you'd have to check if that's correct everywhere `isRef` is used)